packages/firestore/test/integration/api/database.test.ts

```diff
@@ -928,6 +928,61 @@ apiDescribe('Database', persistence => {
     });
   });
 
+  (persistence ? it : it.skip)(
+    'maintains persistence after restarting app',
+    async () => {
+      await withTestDoc(persistence, async docRef => {
+        await docRef.set({ foo: 'bar' });
+        const path = docRef.path;
```

Comment:
There's a lot of noise in this test and the one below that's related to re-recreating the Firestore instance. So much so that it obscures the "interesting" parts of the test.

Is it possible to extract the logic of capturing the details of recreating a Firestore instance? Or maybe make this a shared function near `withTestDoc` that can rely on its implementation details?

Reply:
Decided on not doing so after an in-person conversation, mainly b/c we would only be reducing this by 3 lines.